src/api/app/controllers/person_controller.rb

```diff
@@ -224,22 +223,9 @@ def internal_register
   end
 
   def update_watchlist(user, xml)
```

Comment:
Is something speaking against overwriting `update_watchlist` with the code from `update_new_watchlist`? `update_new_watchlist` could be removed since it would be then unneeded.